docs/how-to/extending/builtin.qmd

```diff
@@ -0,0 +1,134 @@
+---
+freeze: auto
+---
+
+# Reference built-in functions
+
+Functions that aren't exposed in ibis directly can be accessed using the
+`@ibis.udf.scalar.builtin` decorator.
+
+::: {.callout-tip}
+## [Ibis APIs](../../reference/index.qmd) may already exist for your function.
+
+Builtin scalar UDFs are designed to be an escape hatch when Ibis doesn't have
+a defined API for a built-in database function.
+
+See [the reference documentation](../../reference/index.qmd) for existing APIs.
+:::
+
+## DuckDB
+
+Ibis doesn't directly expose many of the DuckDB [text similarity
+functions](https://duckdb.org/docs/sql/functions/char.html#text-similarity-functions).
+Let's expose the `mismatches` API.
+
+```{python}
+from ibis import udf
+
+@udf.scalar.builtin
+def mismatches(left: str, right: str) -> int:
+    ...
+```
+
+The [`...`](https://docs.python.org/3/library/constants.html#Ellipsis) is
+a visual indicator that the function definition is unknown to Ibis.
+
+::: {.callout-note collapse="true"}
+## Ibis does not do anything with the function body.
+
+Ibis will not inspect the function body or otherwise inspect it. Any code you
+write in the function body **will be ignored**.
```

Comment:
Repeat of `inspect` here feels like a typo. Not 100% sure what was intended here, but guessing something like

```suggestion
Ibis will not execute the function body or otherwise inspect it. Any code you
write in the function body **will be ignored**.
```